flocx_market/api/offer_contract_relationship.py

```diff
@@ -6,34 +6,43 @@
 class OfferContractRelationship(Resource):
 
     @classmethod
-    def get(cls, marketplace_offer_id=None, contract_id=None):
-        if (contract_id is None) and (marketplace_offer_id is None):
+    def get(cls):
+        marketplace_offer_id = request.args.get('marketplace_offer_id')
+        contract_id = request.args.get('contract_id')
+        status = request.args.get('status')
+
+        if (contract_id is None) and (marketplace_offer_id is None)\
+                and (status is None):
             return [x.to_dict() for x in
                     ocr.OfferContractRelationship.get_all()]
         ocrs = ocr.OfferContractRelationship.get(marketplace_offer_id,
-                                                 contract_id)
+                                                 contract_id, status)
         if ocrs is None:
```

Comment:
I think this is close, but not quite right; take a look at https://github.com/openstack/ironic/blob/master/ironic/api/controllers/v1/node.py#L1666-L1685 for what may be a cleaner way to pass these filter values.

Reply:
I'm not entirely sure which one are you referring to in this method? Since all these filters are filtering by a single element and requiring the uuid, and in our case we won't know the UUID. Can you explain how did you want that done @tzumainn?

Reply:
So:

* split up `get` and `get_all`
* `get` returns a single offer/contract relationship; it takes the uuid as a parameter.
* `get_all` takes in a single parameter - a dictionary of filters, as the ironic code does. That dictionary of filters is created by the REST API, and unpacked in the database api layer

Reply:
The way this will be used is:

* the provider will query for all of a certain status, and create a contract for each offer/contract relationship. It will also record the relationship's uuid.
* If any update is needed, the provider has the uuid and can use that when querying or making updates.